### DISCUSSION TITLE
RuntimeError: cuDNN error: CUDNN_STATUS_MAPPING_ERROR

### DISCUSSION THREAD

**Discussion by Aayush2003Gupta** [mlcommons/GaNDLF#666] [Q&A] [answered]
**GaNDLF Version**

gandlf 0.0.16
**Desktop (please complete the following information):**
 - Tried both on Ubuntu 18.04 and on Google Colab

**How did you install GaNDLF**
pip install gandlf

**Dataset description**
Working on the BraTS competition 2023

**Describe your question/problem**

Even after preparing the .csv as per the **gandlf_constructCSV.py** and using the .**config_segmentation_brats.yaml** , I get the error
On changing as **image.data = image.data.float()** I get the error:-
 _RuntimeError: cuDNN error: CUDNN_STATUS_MAPPING_ERROR_
![image](https://github.com/mlcommons/GaNDLF/assets/97360205/d8c848db-7f69-4f33-9037-f047bc50c69c)

Also on doing **torch.backends.cudnn.enabled = False**  I end up getting:-
_RuntimeError: CUDA error: CUBLAS_STATUS_NOT_SUPPORTED when calling `cublasGemmEx( handle, opa, opb, m, n, k, &falpha, a, CUDA_R_16F, lda, b, CUDA_R_16F, ldb, &fbeta, c, CUDA_R_16F, ldc, CUDA_R_32F, CUBLAS_GEMM_DFALT_TENSOR_OP)_
![image](https://github.com/mlcommons/GaNDLF/assets/97360205/2b1baf8a-33d8-4151-811e-45e97c29d73d)

Please tell me the error I have tried every possible thing. Thanks a lot for your help!!!!

**Comment by AashrayGupta2003** [comment]
I have the same error popping up. Could you please resolve, any help is deeply awaited, thanks!

**Comment by sarthakpati** [comment]
Thanks for your interest in GaNDLF! 

You mentioned this change:

> On changing as image.data = image.data.float()

Where and why would this be needed? We have performed extensive tests on multi-GPU systems on the current code base and everything works as expected.

How are you ensuring that the current CUDA device is visible to GaNDLF? Before calling `gandlf_run`, the environment variable `CUDA_VISIBLE_DEVICES` needs to be set to the appropriate GPU device.

**Comment by Aayush2003Gupta** [comment]
I changed the function  _def calculate_std_ranges(self, image: torch.Tensor) -> tuple:_ in  _/GANDLF/data/augmentation/blur_enhanced.py_

It was required because the dtype of 1 _image.data_ came out to be int16 which produced the error :-   
_RuntimeError : std and var only support floating point and complex dtypes_
![Untitled_out](https://github.com/mlcommons/GaNDLF/assets/97360205/3593ed79-80e5-4d8c-862c-f7cbfcbaef29)

I had set the **CUDA_VISIBLE_DEVICES** environment variable already as _export CUDA_VISIBLE_DEVICES=0_ , I am  using a Tesla V100 (also tried the same on Google Colab T4 GPU)

**Reply by sarthakpati** [reply]
Hmm, I am unsure why that line would be needed. Regardless, it would be very helpful if you can give me an example using the unit test data so that I can reproduce the error.

You can't add a `.float()` call on its own, unless you are computing on the default device (which is CPU). You need to add it with a `.to` call, like shown in https://github.com/mlcommons/GaNDLF/blob/master/GANDLF/compute/training_loop.py#L98

**Reply by Aayush2003Gupta** [reply, marked as answer]
Thanks a lot the error has been resolved .😀 
I was not giving the full path in the train_data.csv file as mentioned ....

Thanks a lot for your time and support
Aayush Gupta

**Reply by sarthakpati** [reply]
> Thanks a lot the error has been resolved .😀

Thanks for the update! 

So you didn't need to make any changes in the code?

**Reply by Aayush2003Gupta** [reply]
No the code worked perfectly fine .

Thank you 🙏

**Reply by sarthakpati** [reply]
Awesome! Glad to have heard that. 😄 

All the best with your work!
